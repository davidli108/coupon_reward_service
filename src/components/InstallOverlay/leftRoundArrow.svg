<svg width="84" height="93" viewBox="0 0 84 93" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M79.5 89.5C80.6667 77.8333 78.8 50.3 62 33.5C45.2 16.7 18 15.5 6.5 17" stroke="white" stroke-width="7" stroke-linecap="round"/>
<path d="M18 4L5 17L18 30" stroke="white" stroke-width="7" stroke-linecap="round"/>
</svg>
