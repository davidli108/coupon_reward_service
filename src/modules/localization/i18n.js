import i18n from 'i18next';
import parseDomain from 'parse-domain';
import * as R from 'ramda';
import { initReactI18next } from 'react-i18next';

import translationDE from './locales/de/translation';
import translationEN from './locales/en/translation';
import translationFR from './locales/fr/translation';
import translationGB from './locales/gb/translation';

const resources = {
  de: {
    translation: translationDE,
  },
  en: {
    translation: translationEN,
  },
  fr: {
    translation: translationFR,
  },
  gb: {
    translation: translationGB,
  },
};

const Locale = {
  en: 'en',
  de: 'de',
  fr: 'fr',
  gb: 'gb',
};

const localeToDomain = {
  com: Locale.en,
  de: Locale.de,
  fr: Locale.fr,
  'co.uk': Locale.gb,
};

const fallbackLocale = Locale.en;

export const getDomainAttrs = () => {
  const domain = window.location.hostname;
  const protocol = window.location.protocol;

  const parsedDomain = parseDomain(domain);
  const tld = R.propOr(null, 'tld', parsedDomain);
  const enDomain = domain.replace(tld, 'com');
  const enOrigin = `${protocol}//${enDomain}`;

  return {
    domain,
    protocol,
    tld,
    enDomain,
    enOrigin,
  };
};

export const redirectToEnOrigin = () => {
  const domainAttrs = getDomainAttrs();
  window.location.href = domainAttrs.enOrigin;
};

export const getLocale = () => {
  const domainAttrs = getDomainAttrs();
  return localeToDomain[domainAttrs.tld] || fallbackLocale;
};

export const getLocaleConfig = () => {
  return {
    // TODO: enable authentication on all domains after feature support added
    isAuthenticationAvailable: true,
    // TODO: enable follow store on all domains after feature support added
    isFollowStoreAvailable: true,
  };
};

export const initializeI18n = () => {
  // eslint-disable-next-line import/no-named-as-default-member
  i18n.use(initReactI18next).init({
    resources,
    lng: getLocale(),
    fallbackLng: fallbackLocale,
    keySeparator: '.',
    interpolation: {
      escapeValue: false,
    },
  });
};

export default i18n;
