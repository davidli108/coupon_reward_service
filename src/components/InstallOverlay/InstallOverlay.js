// @flow
import React, { useEffect } from 'react';
import styled from 'styled-components';
import leftRoundArrow from './leftRoundArrow.svg';
import Cookie from 'js-cookie';
import moment from 'moment';
import AppConfig from '@config/AppConfig';
import { withTranslation } from 'react-i18next';

type InstallOverlayProps = {
  isActive: boolean,
  callback: Function,
};

const headerHeight = window.outerHeight - window.innerHeight + 'px';
const top =
  navigator.platform === 'Win32'
    ? `calc(260px - ${headerHeight})`
    : `calc(210px - ${headerHeight})`;

const InstallOverlay = ({ t, isActive, callback }: InstallOverlayProps) => {
  const windowProps = {
    toolbar: 'no',
    location: 'no',
    directories: 'no',
    status: 'no',
    menubar: 'no',
    scrollbars: 'no',
    resizable: 'no',
    width: '1100',
    height: '700',
    left: window.screenX,
    top: 0,
  };

  const parsedWindowProps = () => {
    const data = [];
    Object.keys(windowProps).forEach(key => {
      data.push(`${key}=${windowProps[key]}`);
    });

    return data.join(',');
  };

  useEffect(() => {
    if (isActive) {
      const popup = window.open(
        AppConfig.extension.url,
        'extensionWindow',
        parsedWindowProps(),
      );

      const focusInterval = setInterval(() => {
        popup.focus();

        if (popup.closed) {
          clearInterval(focusInterval);
          Cookie.set('installProcessed', true, {
            expires: moment()
              .add(1, 'days')
              .toDate(),
          });

          callback();
        }
      }, 100);
    }
  }, [isActive]);

  return (
    <InstallOverlay.Wrapper isActive={isActive}>
      <InstallOverlay.Container top={top}>
        <img src={leftRoundArrow} alt={t('coupons.installExtension.action')} />
        <InstallOverlay.Step>
          <h3>{t('coupons.installExtension.step', { step: '1' })}</h3>
          <p>{t('coupons.installExtension.step1text')}</p>
        </InstallOverlay.Step>

        <InstallOverlay.Step>
          <h3>{t('coupons.installExtension.step', { step: '2' })}</h3>
          <p>{t('coupons.installExtension.step2text')}</p>
        </InstallOverlay.Step>
      </InstallOverlay.Container>
    </InstallOverlay.Wrapper>
  );
};

InstallOverlay.Wrapper = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1050;
  background-color: rgba(0, 0, 0, 0.9);
  color: #fff;
`;

InstallOverlay.Container = styled.div`
  position: absolute;
  left: 1130px;
  top: ${({ top }) => top};
  width: 320px;

  > img {
    display: block;
    width: 75px;
    height: auto;
    position: relative;
    margin: 0 0 17px;
  }
`;

InstallOverlay.Step = styled.div`
  padding: 0 0 0 36px;
  margin: 0 0 30px;

  h3 {
    margin: 0 0 5px;
    font-weight: bold;
    font-size: 28px;
  }

  p {
    margin: 0;
    font-size: 24px;
  }
`;

export default withTranslation()(InstallOverlay);
