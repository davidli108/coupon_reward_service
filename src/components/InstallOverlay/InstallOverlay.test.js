import React from 'react';
import InstallOverlay from './InstallOverlay';
import renderer from 'react-test-renderer';
import { initializeI18n } from '../../modules/localization/i18n';
import 'jest-styled-components';

initializeI18n();

it('renders correctly', () => {
  const tree = renderer
    .create(<InstallOverlay isActive={true} callback={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
