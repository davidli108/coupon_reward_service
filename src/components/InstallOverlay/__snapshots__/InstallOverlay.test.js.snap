// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="InstallOverlay__Wrapper-sc-10gjagm-0 cxFTog"
>
  <div
    className="InstallOverlay__Container-sc-10gjagm-1 cWQicq"
  >
    <img
      alt="coupons.installExtension.action"
      src="leftRoundArrow.svg"
    />
    <div
      className="InstallOverlay__Step-sc-10gjagm-2 bYzkZP"
    >
      <h3>
        coupons.installExtension.step1
      </h3>
      <p>
        coupons.installExtension.step1text
      </p>
    </div>
    <div
      className="InstallOverlay__Step-sc-10gjagm-2 bYzkZP"
    >
      <h3>
        coupons.installExtension.step2
      </h3>
      <p>
        coupons.installExtension.step2text
      </p>
    </div>
  </div>
</div>
`;
